src/components/ConfigEditor/QueryConfig.tsx

```diff
@@ -0,0 +1,89 @@
+import { AdxDataSourceOptions, AdxDataSourceSecureOptions, EditorMode } from 'types';
+
+interface QueryConfigProps
+  extends DataSourcePluginOptionsEditorProps<AdxDataSourceOptions, AdxDataSourceSecureOptions> {
+  updateJsonData: <T extends keyof AdxDataSourceOptions>(fieldName: T, value: AdxDataSourceOptions[T]) => void;
+}
+
+const QueryConfig: React.FC<QueryConfigProps> = ({ options, updateJsonData }) => {
+  const { jsonData } = options;
+
+  const dataConsistencyOptions: Array<{ value: string; label: string }> = [
+    { value: 'strongconsistency', label: 'Strong' },
+    { value: 'weakconsistency', label: 'Weak' },
+  ];
+
+  const editorModeOptions: Array<SelectableValue<EditorMode>> = [
+    { value: EditorMode.Visual, label: 'Visual' },
+    { value: EditorMode.Raw, label: 'Raw' },
+  ];
+
+  return (
+    <FieldSet label="Query Optimizations">
+      <InlineField label="Query timeout" labelWidth={26} tooltip="This value controls the client query timeout.">
+        <Input
+          value={jsonData.queryTimeout}
+          id="adx-query-timeout"
+          placeholder="30s"
+          width={18}
+          onChange={(ev: React.ChangeEvent<HTMLInputElement>) => updateJsonData('queryTimeout', ev.target.value)}
+        />
+      </InlineField>
+
+      <InlineField
+        label="Use dynamic caching"
+        labelWidth={26}
+        tooltip="By enabling this feature Grafana will dynamically apply cache settings on a per query basis and the default cache max age will be ignored.<br /><br />For time series queries we will use the bin size to widen the time range but also as cache max age."
+      >
+        <InlineSwitch
+          value={jsonData.dynamicCaching}
+          id="adx-caching"
+          transparent={false}
+          onChange={(ev: React.ChangeEvent<HTMLInputElement>) => updateJsonData('dynamicCaching', ev.target.checked)}
+        />
+      </InlineField>
+
+      <InlineField
+        label="Cache max age"
+        labelWidth={26}
+        tooltip="By default the cache is disabled. If you want to enable the query caching please specify a max timespan for the cache to live."
+      >
+        <Input
+          value={jsonData.cacheMaxAge}
+          id="adx-cache-age"
+          placeholder="0m"
+          width={18}
+          onChange={(ev: React.ChangeEvent<HTMLInputElement>) => updateJsonData('cacheMaxAge', ev.target.value)}
+        />
+      </InlineField>
+
+      <InlineField label="Data consistency" labelWidth={26}>
+        <Select
+          options={dataConsistencyOptions}
+          value={dataConsistencyOptions.find(v => v.value === jsonData.dataConsistency)}
+          onChange={(change: SelectableValue<string>) =>
+            updateJsonData('dataConsistency', change.value ? change.value : dataConsistencyOptions[0].value)
+          }
+          isClearable={false}
+          width={18}
+        />
+      </InlineField>
+
+      <InlineField label="Default editor mode" labelWidth={26}>
```

Comment:
I deliberately didnt add the tooltip because I didnt find it added anything?

> The default editor mode is Visual. If you need to change the default editor mode choose the desired value.

"if you need to change the value, change the value" - I don't think this adds anything?

Reply:
okay :)